### ISSUE TITLE
ncp-config not working through SSH

### ISSUE THREAD

**Issue by Sweenu** [nextcloud/nextcloudpi#494]
I have been having a problem with ncp-config: 
It works flawlessly when I plug my raspberryPi to a display and launch it from there. Works great as well through the web-ui. However, for some reasons, I can't get it to work through ssh.

The tui opens and I can select different items but it will always stop at step n-1 with n being the last step. For example, when I select 'letsencrypt' it will give me the warning (step 1) but after selecting yes, it just doesn't do anything and goes back to the main menu instead of going to the second and final step of entering the domain name and mail.
For items that have only one step, it just doesn't do anything.

I have no ideas where it comes from and if there are any logs I can look at for the ncp-config program.
I guess it has to do with ssh configuration?

Output of `sudo ncp-report` :
<details>
<summary>NextCloudPlus diagnostics</summary>

```

NextCloudPlus version  v0.54.1
NextCloudPlus image    NextCloudPlus_04-21-18
distribution           Raspbian GNU/Linux 9 \n \l
automount              no
USB devices            none
datadir                /var/www/nextcloud/data
data in SD             yes
data filesystem        ext2/ext3
data disk usage        1.9G/29G
rootfs usage           1.9G/29G
swapfile               /var/swap
Nextcloud check        ok
Nextcloud version      13.0.1.1
HTTPD service          up
PHP service            up
MariaDB service        up
Redis service          up
Postfix service        up
internet check         ok
port check 80          open
port check 443         open
IP                     192.168.1.98
gateway                192.168.1.1
interface              eth0
certificates           none
NAT loopback           no
uptime                 10:44

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": {
            "0": "localhost",
            "5": "nextcloudplus.local",
            "1": "192.168.1.98"
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbtype": "mysql",
        "version": "13.0.1.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "mail_smtpmode": "php",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "overwriteprotocol": "https",
        "loglevel": "2",
        "log_type": "file"
    }
}

```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Sat Apr 28 08:26:37.819450 2018] [mpm_event:notice] [pid 799:tid 1992093696] AH00493: SIGUSR1 received.  Doing graceful restart
[Sat Apr 28 08:26:37.865353 2018] [ssl:warn] [pid 799:tid 1992093696] AH01909: localhost:4443:0 server certificate does NOT include an ID which matches the server name
[Sat Apr 28 08:26:37.865712 2018] [ssl:error] [pid 799:tid 1992093696] AH02217: ssl_stapling_init_cert: can't retrieve issuer certificate! [subject: CN=raspberrypi / issuer: CN=raspberrypi / serial: AE8EA77112056528 / notbefore: Apr 20 14:05:43 2018 GMT / notafter: Apr 17 14:05:43 2028 GMT]
[Sat Apr 28 08:26:37.865777 2018] [ssl:error] [pid 799:tid 1992093696] AH02604: Unable to configure certificate localhost:4443:0 for stapling
[Sat Apr 28 08:26:37.866586 2018] [ssl:warn] [pid 799:tid 1992093696] AH01909: localhost:443:0 server certificate does NOT include an ID which matches the server name
[Sat Apr 28 08:26:37.866852 2018] [ssl:error] [pid 799:tid 1992093696] AH02217: ssl_stapling_init_cert: can't retrieve issuer certificate! [subject: CN=raspberrypi / issuer: CN=raspberrypi / serial: AE8EA77112056528 / notbefore: Apr 20 14:05:43 2018 GMT / notafter: Apr 17 14:05:43 2028 GMT]
[Sat Apr 28 08:26:37.866874 2018] [ssl:error] [pid 799:tid 1992093696] AH02604: Unable to configure certificate localhost:443:0 for stapling
[Sat Apr 28 08:26:38.000321 2018] [mpm_event:notice] [pid 799:tid 1992093696] AH00489: Apache/2.4.25 (Raspbian) OpenSSL/1.0.2l configured -- resuming normal operations
[Sat Apr 28 08:26:38.000392 2018] [core:notice] [pid 799:tid 1992093696] AH00094: Command line: '/usr/sbin/apache2'
[Sat Apr 28 08:27:43.723946 2018] [mpm_event:notice] [pid 799:tid 1992093696] AH00493: SIGUSR1 received.  Doing graceful restart
[Sat Apr 28 08:27:43.767902 2018] [ssl:warn] [pid 799:tid 1992093696] AH01909: localhost:4443:0 server certificate does NOT include an ID which matches the server name
[Sat Apr 28 08:27:43.768267 2018] [ssl:error] [pid 799:tid 1992093696] AH02217: ssl_stapling_init_cert: can't retrieve issuer certificate! [subject: CN=raspberrypi / issuer: CN=raspberrypi / serial: AE8EA77112056528 / notbefore: Apr 20 14:05:43 2018 GMT / notafter: Apr 17 14:05:43 2028 GMT]
[Sat Apr 28 08:27:43.768291 2018] [ssl:error] [pid 799:tid 1992093696] AH02604: Unable to configure certificate localhost:4443:0 for stapling
[Sat Apr 28 08:27:43.769081 2018] [ssl:warn] [pid 799:tid 1992093696] AH01909: localhost:443:0 server certificate does NOT include an ID which matches the server name
[Sat Apr 28 08:27:43.769346 2018] [ssl:error] [pid 799:tid 1992093696] AH02217: ssl_stapling_init_cert: can't retrieve issuer certificate! [subject: CN=raspberrypi / issuer: CN=raspberrypi / serial: AE8EA77112056528 / notbefore: Apr 20 14:05:43 2018 GMT / notafter: Apr 17 14:05:43 2028 GMT]
[Sat Apr 28 08:27:43.769369 2018] [ssl:error] [pid 799:tid 1992093696] AH02604: Unable to configure certificate localhost:443:0 for stapling
[Sat Apr 28 08:27:44.000403 2018] [mpm_event:notice] [pid 799:tid 1992093696] AH00489: Apache/2.4.25 (Raspbian) OpenSSL/1.0.2l configured -- resuming normal operations
[Sat Apr 28 08:27:44.000476 2018] [core:notice] [pid 799:tid 1992093696] AH00094: Command line: '/usr/sbin/apache2'
[Sat Apr 28 08:28:29.135826 2018] [authz_core:error] [pid 11792:tid 1756361776] [client 66.133.109.36:60682] AH01630: client denied by server configuration: /var/www/ncp-web/.well-known, referer: http://cloud.inec.fr/.well-known/acme-challenge/h6UCLjNObJyxc2oHfPM8iE8RKOzq7RMOPiDg74AQkbE
[Sat Apr 28 10:37:46.643215 2018] [authz_core:error] [pid 11792:tid 1984586800] [client 137.226.113.20:51022] AH01630: client denied by server configuration: /var/www/ncp-web/

```
</details>

<details>
<summary>Database logs</summary>

```

2018-04-28  0:00:42 1978868544 [Note] /usr/sbin/mysqld: Shutdown complete

2018-04-28  0:02:03 1989001216 [Note] InnoDB: Using mutexes to ref count buffer pool pages
2018-04-28  0:02:03 1989001216 [Note] InnoDB: The InnoDB memory heap is disabled
2018-04-28  0:02:03 1989001216 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2018-04-28  0:02:03 1989001216 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
2018-04-28  0:02:03 1989001216 [Note] InnoDB: Compressed tables use zlib 1.2.8
2018-04-28  0:02:03 1989001216 [Note] InnoDB: Using Linux native AIO
2018-04-28  0:02:03 1989001216 [Note] InnoDB: Using generic crc32 instructions
2018-04-28  0:02:03 1989001216 [Note] InnoDB: Initializing buffer pool, size = 128.0M
2018-04-28  0:02:03 1989001216 [Note] InnoDB: Completed initialization of buffer pool
2018-04-28  0:02:03 1989001216 [Note] InnoDB: Highest supported file format is Barracuda.
2018-04-28  0:02:05 1989001216 [Note] InnoDB: 128 rollback segment(s) are active.
2018-04-28  0:02:05 1989001216 [Note] InnoDB: Waiting for purge to start
2018-04-28  0:02:05 1989001216 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.35-80.0 started; log sequence number 3530745
2018-04-28  0:02:05 1447031616 [Note] InnoDB: Dumping buffer pool(s) not yet started
2018-04-28  0:02:05 1989001216 [Note] Plugin 'FEEDBACK' is disabled.
2018-04-28  0:02:05 1989001216 [Note] Server socket created on IP: '127.0.0.1'.
2018-04-28  0:02:06 1989001216 [Note] /usr/sbin/mysqld: ready for connections.
Version: '10.1.23-MariaDB-9+deb9u1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  Raspbian 9.0

```
</details>

<details>
<summary>Nextcloud logs</summary>

```

{"reqId":"7oWusGz5voZTr24iwFFe","level":4,"time":"2018-04-21T11:15:42+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"LhAOWosookHQwkgLRBkt","level":4,"time":"2018-04-21T11:30:16+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"OICTazOGVM1n4O0WYwnv","level":4,"time":"2018-04-21T11:45:17+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"vbo3G5olGIJiqpoIMbYf","level":4,"time":"2018-04-21T12:00:16+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"6IlOKGqwMDGz0yMJsRCq","level":4,"time":"2018-04-21T12:15:19+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"6MDixblPJh4CCmuIwikE","level":4,"time":"2018-04-21T12:30:13+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"v9WoqTRYCgFFcD9NlLPk","level":4,"time":"2018-04-21T12:45:14+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"z1FChrOqoErkyM9RIrAq","level":4,"time":"2018-04-21T13:00:16+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"KqjCr9XE1qe5qXFprWJW","level":4,"time":"2018-04-21T13:15:17+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"5tgUofBog3BqSxEj2PSK","level":4,"time":"2018-04-21T13:30:18+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":""}
{"reqId":"88TuTxiXVVGuEVmMRunT","level":4,"time":"2018-04-21T13:45:19+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Not installed","userAgent":"--","version":"13.0.1.1"}
{"reqId":"s5FreASCSnqv6UHHWcCW","level":3,"time":"2018-04-27T22:50:46+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"Redis::connect(): connect() failed: No such file or directory at \/var\/www\/nextcloud\/lib\/private\/RedisFactory.php#84","userAgent":"--","version":"13.0.1.1"}

```
</details>

**Comment by Sweenu** [comment]
Okay so I know where the problem comes from. It wasn't ssh but my terminal.
I'm using termite as a terminal emulator and apparently the interface has some kind of bug with it because it works fine with xterm.

Do you have any idea what could cause that and how I could fix it?

**Comment by nachoparker** [comment]
no idea, you could try to replace [this line](https://github.com/nextcloud/nextcloudpi/blob/master/etc/library.sh#L153) for a simple `clear`

**Comment by Sweenu** [comment]
That didn't work but I found the problem (and solution, kinda).

The problem is that for ncurses based app to fully work on termite, it needs `/lib/terminfo/t/termite` and `$TERM` set to `termite` on the host.

New versions of ncurses include `/lib/terminfo/t/termite` so an ncurses update would solve part of the problem. The rest of the problem comes from termite which would need to accept [this pull request](https://github.com/thestinger/termite/pull/591) I just made in order to use ncurses' terminfo file instead of their own they were using to make up for ncurses not having one until recently.
